File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int tar) {
        
        
        int sx = Solution.bs(nums ,tar) ; 
        if( sx == nums.length || nums[sx] !=tar) return new int[]{-1,-1} ;
        return new int[]{sx ,Solution.bs(nums,tar+1) -1} ; 
    }
    private static int bs(int[] nums, int tar){ 
    
        int l =0 , h = nums.length ; 
        while(l < h ){
            int m = l+((h-l)>>1) ;

            if(nums[m] < tar) l = m+1 ;
            else h = m ; 
        }
        return l ; 
  
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], tar: int) -> List[int]:
        
        if not nums: return [-1,-1]
        nl = len(nums)
        lk = bisect.bisect_left(nums,tar)
        lk = lk if 0<=lk <nl and nums[lk]==tar else -1
        rk = bisect.bisect_right(nums,tar)
        print(rk)
        rk = rk -1 if  0<=rk-1<nl and nums[rk-1] ==tar else -1 
        return [lk,rk ]